finagle-base-http/src/main/scala/com/twitter/finagle/http/TraceInfo.scala

```diff
@@ -12,6 +12,47 @@ private object TraceInfo {
   }
 
   def letTraceIdFromRequestHeaders[R](request: Request)(f: => R): R = {
+    // rather than rewrite all this to handle more I'm just shimming
+    // in a check for the new trace b3 header and if it exists I'm
+    // extracting out its components to be handled by the existing
+    // code.
+    if (request.headerMap.contains(Header.TraceContext)) {
+      def handleOne(headers: HeaderMap,value: String): Unit =
+        value match {
+          case "d" => headers.set(Header.Flags,"1")
+          case "1" => headers.set(Header.Sampled,"1")
+          case _ => ()
+        }
+      def handleTwo(headers: HeaderMap, a: String,b: String): Unit = {
```

Comment:
I think we can make the names more about semantics than about syntax and it'll be easier to use.  can we call this `setIdentity` or something like that?  and for `setOne` can we call it `setDebugMode`?